app/org/sagebionetworks/bridge/services/AuthenticationServiceImpl.java

```diff
@@ -246,14 +257,23 @@ private UserSession getSessionFromAccount(final Study study, final Account accou
 
         final String healthCode = getHealthCode(study, account);
         user.setHealthCode(healthCode);
-
+        
         user.setSharingScope(optionsService.getEnum(healthCode, SHARING_SCOPE, SharingScope.class));
         user.setDataGroups(optionsService.getStringSet(healthCode, DATA_GROUPS));
-        
-        user.setSignedMostRecentConsent(consentService.hasUserSignedActiveConsent(study, user));
-        user.setConsent(consentService.hasUserConsentedToResearch(study, user));
 
+        // now that we know more about this user, we can expand on the request context.
+        context = new ScheduleContext.Builder()
+                .withContext(context)
+                .withHealthCode(healthCode)
+                .withUserDataGroups(user.getDataGroups())
+                .withStudyIdentifier(study.getIdentifier()) // probably already set
+                .build();
```

Comment:
Yes, absolutely. Once I turn to the schedules PR which also depends on some of the same information (which is being extracted out to the Criteria and CriteriaUtils classes), I'll reduce and/or change this. I'm not sure at this time what that will look like because we're still refactoring this. I think there's a CriteriaContext object with min/max app version\, clientInfo and data groups, and schedule context includes additional information. The consent matching may just use CriteriaContext.